conan/tools/system/package_manager.py

```diff
@@ -0,0 +1,175 @@
+class SystemPackageManagerTool(object):
+    install_methods = ["install", "update"]
+    mode_check = "check"
+    tool_name = None
+    install_command = ""
+    update_command = ""
+    check_command = ""
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._active_tool = self._conanfile.conf["tools.system.package_manager:tool"]
+        self._sudo = self._conanfile.conf["tools.system.package_manager:sudo"]
+        self._sudo_askpass = self._conanfile.conf["tools.system.package_manager:sudo_askpass"]
+        self._mode = self._conanfile.conf["tools.system.package_manager:mode"] or self.mode_check
+        self._arch = self._conanfile.settings.get_safe("arch")
+        self._arch_names = {}
+        self._arch_separator = ""
+
+    def get_package_name(self, package):
+        # TODO: should we only add the arch if cross-building?
+        if self._arch in self._arch_names:
+            return "{}{}{}".format(package, self._arch_separator,
+                                   self._arch_names.get(self._arch))
+        return package
+
+    @property
+    def sudo_str(self):
+        sudo = "sudo " if self._sudo else ""
+        askpass = "-A " if self._sudo and self._sudo_askpass else ""
+        return "{}{}".format(sudo, askpass)
+
+    def method_decorator(wrapped):
+        def wrapper(self, *args, **kwargs):
+            method_name = wrapped.__name__
+            if self._active_tool == self.__class__.tool_name:
+                if method_name in self.install_methods and self._mode == self.mode_check:
+                    self._conanfile.output.warn("Can't {}. Please update packages manually or set "
+                                                "'tools.system.package_manager:mode' to "
+                                                "'install'".format(method_name))
+                else:
+                    return wrapped(self, *args, **kwargs)
+            else:
+                self._conanfile.output.warn("Ignoring call to {}.{}. Set "
```

Comment:
This seems will be very noisy when a recipe adds a few managers for different platforms, all the others not active will be constantly warning? Maybe better remove this warning.